app/assets/javascripts/discourse/app/mixins/uppy-upload.js

```diff
@@ -143,6 +171,40 @@ export default Mixin.create({
     });
   },
 
+  _useS3Uploads() {
+    this.set("usingS3Uploads", true);
+    this.uppyInstance.use(AwsS3, {
+      getUploadParameters: (file) => {
+        const data = { file_name: file.name, type: this.type };
+
+        // the sha1 checksum is set by the UppyChecksum plugin, except
+        // for in cases where the browser does not support the required
+        // crypto mechanisms or an error occurs. it is an additional layer
+        // of security, and not required.
+        if (file.meta.sha1_checksum) {
+          data.metadata = { "sha1-checksum": file.meta.sha1_checksum };
+        }
+
+        return ajax(getUrl("/uploads/generate-presigned-put"), {
```

Comment:
I think we need to consider what happens if this request returns an error.

Reply:
Good catch...I forgot the `catch`...okay that's enough dad jokes for one day